forest/sycamore/sycamore_main.py

```diff
@@ -1,48 +1,92 @@
         Timezone of study. This defaults to 'America/New_York'
-    """
+    interventions_filepath(str):
+        filepath where interventions json file is.
 
+    """
+    os.makedirs(output_folder, exist_ok=True)
+    if participant_ids is None:
+        participant_ids = [u for u in os.listdir(
+            study_folder) if not u.startswith('.') and u != 'registry']
     # Read, aggregate and clean data
     if config_path is None:
         print('No config file provided. Skipping some summary outputs.')
-        agg_data = aggregate_surveys_no_config(study_dir, study_tz)
-
+        agg_data = aggregate_surveys_no_config(
+            study_folder, tz_str, participant_ids)
+        if agg_data.shape[0] == 0:
+            print("Error: No survey data found")
+            return(True)
     else:
-        agg_data = aggregate_surveys_config(study_dir, config_path, study_tz)
+        agg_data = aggregate_surveys_config(study_folder, config_path,
+                                            tz_str, participant_ids)
+        if agg_data.shape[0] == 0:
+            print("Error: No survey data found")
+            return True
         # Create changed answers detail and summary
-        ca_detail, ca_summary = agg_changed_answers_summary(study_dir, config_path, agg_data, study_tz)
-        ca_detail.to_csv(os.path.join(output_dir, 'answers_data.csv'), index=False)
-        ca_summary.to_csv(os.path.join(output_dir, 'answers_summary.csv'), index=False)
+        ca_detail, ca_summary = agg_changed_answers_summary(
+            study_folder, config_path, agg_data, tz_str)
+        ca_detail.to_csv(os.path.join(output_folder, 'answers_data.csv'),
+                         index=False)
+        ca_summary.to_csv(os.path.join(output_folder, 'answers_summary.csv'),
+                          index=False)
         if time_start is not None and time_end is not None:
             # Create survey submits detail and summary
-            ss_detail, ss_summary = survey_submits(study_dir, config_path, time_start, time_end, beiwe_ids, agg_data,
-                                                   study_tz)
-            ss_detail.to_csv(os.path.join(output_dir, 'submits_data.csv'), index=False)
-            ss_summary.to_csv(os.path.join(output_dir, 'submits_summary.csv'), index=False)
+            all_interventions_dict = get_all_interventions_dict(
+                interventions_filepath)
+            ss_detail, ss_summary = survey_submits(
+                config_path, time_start, time_end,
+                participant_ids, agg_data, all_interventions_dict)
+            if ss_summary.shape[0] > 0:
+                ss_detail.to_csv(
+                    os.path.join(
+                        output_folder,
+                        'submits_data.csv'),
+                    index=False)
+                ss_summary.to_csv(os.path.join(output_folder,
+                                               'submits_summary.csv'),
+                                  index=False)
```

Comment:
More readable:
```
                ss_summary.to_csv(os.path.join(
                    output_folder, 'submits_summary.csv'
                ), index=False)
```

Reply:
Updated this. I think it at least makes more sense for me to put the os.path.join on its own line so I can understand what the filepath will be